flink-yarn/src/main/java/org/apache/flink/yarn/cli/FallbackYarnSessionCli.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.yarn.cli;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.client.cli.AbstractCustomCommandLine;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.DeploymentOptions;
+import org.apache.flink.yarn.configuration.YarnConfigOptions;
+import org.apache.flink.yarn.configuration.YarnDeploymentTarget;
+import org.apache.flink.yarn.executors.YarnJobClusterExecutor;
+import org.apache.flink.yarn.executors.YarnSessionClusterExecutor;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+/**
+ * A stub Yarn Command Line to throw an exception with the correct
+ * message when the {@code HADOOP_CLASSPATH} is not set.
+ */
+@Internal
+public class FallbackYarnSessionCli extends AbstractCustomCommandLine {
+
+	public static final String ID = "yarn-cluster";
+
+	private final Option applicationId;
+
+	public FallbackYarnSessionCli(Configuration configuration) {
+		super(configuration);
+		applicationId = new Option("yid", "yarnapplicationId", true, "Attach to running YARN session");
+	}
+
+	@Override
+	public void addGeneralOptions(Options baseOptions) {
+		super.addGeneralOptions(baseOptions);
+		baseOptions.addOption(applicationId);
+	}
+
+	@Override
+	public boolean isActive(CommandLine commandLine) {
+		if (originalIsActive(commandLine)) {
+			throw new IllegalStateException(YarnDeploymentTarget.ERROR_MESSAGE);
+		}
+		return false;
+	}
+
+	private boolean originalIsActive(CommandLine commandLine) {
```

Comment:
So we are intentional to ignore the Yarn properties file here. Right?

Reply:
Yes, the reason is that:
1) we would have to copy more code
2) I would assume that if we have the properties (hidden) file, it means that we created the Yarn cluster using the session CLI, so we already have hadoop in the classpath.

(In fact the solution here is already fragile, as whoever touches the `isActive()` should know to keep both methods in sync)

Do you agree @wangyang0918 ?

Reply:
Yeah, i agree with you that we do not need to add the logics for Yarn magic properties file here.

Reply:
Yeah, i agree with you that we do not need to add the logics for Yarn magic properties file here.